### PR TITLE
Comsat testing

### PR DESCRIPTION
**Pull request by ekrylovich**

### CODE REVIEW

File: daemon/src/org/eclipse/iofog/process_manager/DockerUtil.java [hunk 1]
```diff
@@ -342,27 +338,35 @@ public boolean isNetworkModeEqual(InspectContainerResponse inspectInfo, Element
...
-		return comparePortMapping(elementPorts, ports.getBindings());
+		List<PortMapping> elementPorts = getElementPorts(element);
+		List<PortMapping> containerPorts = getContainerPorts(inspectInfo);
+		return isPortMappingListsEqual(elementPorts, containerPorts);
+
 	}
 
-	private boolean comparePortMapping(List<PortMapping> elementPorts, Map<ExposedPort, Ports.Binding[]> portBindings) {
+	private boolean isPortMappingListsEqual(List<PortMapping> elementPorts, List<PortMapping> containerPorts) {
+		return elementPorts.size() == containerPorts.size()
```
Comment (ekrylovich) [inline daemon/src/org/eclipse/iofog/process_manager/DockerUtil.java]: equal?

File: daemon/src/org/eclipse/iofog/process_manager/ProcessManager.java [hunk 2]
```diff
@@ -118,11 +115,13 @@ public void update() {
 							element.setContainerIpAddress("0.0.0.0");
 						}
 						ElementStatus status = docker.getElementStatus(container.getId());
+						if (status.getStatus() == ElementState.REMOVED) {
+							latestElements.remove(element);
```
Comment (ekrylovich) [inline daemon/src/org/eclipse/iofog/process_manager/ProcessManager.java]: reverse engineering

File: daemon/src/org/eclipse/iofog/process_manager/DockerUtil.java [hunk 3]
```diff
@@ -252,20 +253,30 @@ private long getStartedTime(String startedTime) {
...
+	}
+
+	private ElementStatus getElementStatusOfExistingContainer(String containerId) {
 		InspectContainerResponse inspectInfo = dockerClient.inspectContainerCmd(containerId).exec();
-		ContainerState containerState = inspectInfo.getState();
-		ElementStatus result = new ElementStatus();
-		if (containerState != null) {
-			if (containerState.getStartedAt() != null) {
-				result.setStartTime(getStartedTime(containerState.getStartedAt()));
+		ContainerState status = inspectInfo.getState();
```
Comment (elukashick) [inline daemon/src/org/eclipse/iofog/process_manager/DockerUtil.java]: maybe rename to containerState

File: daemon/src/org/eclipse/iofog/process_manager/DockerUtil.java [hunk 4]
```diff
@@ -372,9 +384,22 @@ public boolean hasContainerWithElementId(String elementId) {
...
 	}
 
+	/**
+	 * returns whether the {@link Container} exists or not
+	 * preferable to perform a check by elementId
+	 *
+	 * @param containerId - id of {@link Element}
+	 * @return boolean true if exists and false in other case
+	 */
+	public boolean hasContainerWithContainerId(String containerId) {
```
Comment (elukashick) [inline daemon/src/org/eclipse/iofog/process_manager/DockerUtil.java]: is it still used?

File: daemon/src/org/eclipse/iofog/process_manager/ProcessManager.java [hunk 5]
```diff
@@ -115,27 +118,49 @@ public void update() {
...
 			} catch (Exception ex) {
 				logWarning(ex.getMessage());
 			}
-			elementManager.setCurrentElements(latestElements);
+			elementManager.setCurrentElements(elementsWithExistingContainers);
 		}
 	};
 
+	private boolean shouldContainerBeUpdated(ElementStatus status, String containerId, Element element) {
+		boolean isNotRunning = !(ElementState.RUNNING == status.getStatus());
```
Comment (ekrylovich) [inline daemon/src/org/eclipse/iofog/process_manager/ProcessManager.java]: status

**Pull request closed without merging**